test/integration/http2_integration_test.cc

```diff
@@ -178,6 +178,39 @@ TEST_P(Http2IntegrationTest, Trailers) { testTrailers(1024, 2048); }
 
 TEST_P(Http2IntegrationTest, TrailersGiantBody) { testTrailers(1024 * 1024, 1024 * 1024); }
 
+TEST_P(Http2IntegrationTest, GrpcRequestTimeout) {
+  config_helper_.addConfigModifier(
+      [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm)
+          -> void {
+        auto* route_config = hcm.mutable_route_config();
+        auto* virtual_host = route_config->mutable_virtual_hosts(0);
+        auto* route = virtual_host->mutable_routes(0);
+        route->mutable_route()->set_use_grpc_timeout(true);
+      });
+  initialize();
+
+  // Envoy will close some number of connections when request times out.
+  // Make sure they don't cause assertion failures when we ignore them.
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
```

Comment:
Please make sure you run this new tests in a loop 1000 times using bazel just to make sure there are no flakes.

Reply:
Did not know how else to disable other tests in the file, so I disabled them (`#if 0`...`#endif`) manually, then ran:
```
bazel test --runs_per_test=1000 //test/integration:http2_integration_test
```

With these results:

```
INFO: 1001 processes, linux-sandbox.
INFO: Build completed successfully, 1002 total actions
//test/integration:http2_integration_test                                PASSED in 2.9s
  Stats over 1000 runs: max = 2.9s, min = 2.7s, avg = 2.8s, dev = 0.0s
```

Is this what you wanted (never done anything like this before with Bazel)?